internal/logging/logging.go

```diff
@@ -0,0 +1,185 @@
+package logging
+
+import (
+	"fmt"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"os"
+	"sync"
+	"time"
+)
+
+// Logging implements logging using: a switch to disable logging on Fatal log, default logger and level,
+// and stored child loggers and their options.
+type Logging struct {
+	// enabled implements a switch to disable logging on Fatal log.
+	enabled *atomic.Bool
+	// level defines default log-level.
+	level zap.AtomicLevel
+	// output defines default log-level.
+	output string
+	// logger defines default logger.
+	logger *zap.SugaredLogger
+	// encoder stores zapcore.Encoder; used to build the logger
+	encoder zapcore.Encoder
+	// encoder stores zapcore.WriteSyncer; used to build the logger
+	syncer zapcore.WriteSyncer
+	mu     sync.Mutex
+	// loggers stores named child loggers.
+	loggers map[string]*zap.SugaredLogger
+	options Options
+}
+
+// Options stores a `child-logger-name: level` pair mapping.
+type Options map[string]string
+
+// NewLogging returns a initialized Logging
+// using level and options from configuration file.
+func NewLogging(level string, output string, options Options) *Logging {
+	var lvl zapcore.Level
+	if err := lvl.Set(level); err != nil {
+		panic(err)
+	}
+	atom := zap.NewAtomicLevelAt(lvl)
+	enabled := atomic.NewBool(true)
+	encoder, syncer := selectEncoderSyncer(output)
+	logger := zap.New(zapcore.NewCore(
+		encoder,
+		syncer,
+		zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+			// Always log fatals, but allow other levels to be completely disabled.
+			return zap.FatalLevel.Enabled(lvl) || enabled.Load() && atom.Enabled(lvl)
+		}),
+	))
+
+	return &Logging{
+		enabled: enabled,
+		level:   atom,
+		output:  output,
+		logger:  logger.Sugar(),
+		encoder: encoder,
+		syncer:  syncer,
+		mu:      sync.Mutex{},
```

Comment:
You don't need that initialization. The zero value of `sync.Mutex` is already usable.